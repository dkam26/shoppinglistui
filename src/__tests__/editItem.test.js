import React from 'react';
import { shallow, mount } from 'enzyme';
import {configure} from 'enzyme';
import EditItem  from '../components/edit.item.component';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import  toJson,{ shallowToJson } from 'enzyme-to-json';
import LocalStorageMock from '../setupTests';
import ReactDOM from 'react-dom';
import {Button,Form} from 'semantic-ui-react';
configure({ adapter: new Adapter() });
describe('edit item component', () => {
    const props={
        getItem:()=>{},
        componentWillMount:()=>{},
        getLists:()=>{},
        onSubmit:()=>{},
       
        match:{
            params:{}
            }
    }
    const wrapper =  shallow(<EditItem {...props} />);
    it('should run', () => {
        expect(wrapper.exists(<div></div>)).toBe(true);
    });
    it('should render correctly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        });
    it('should edit item',()=>{
            expect(wrapper.find(<Button id='editItem'></Button>));
    
        });
   
    it('should ve the following methods',()=>{
            wrapper.instance().componentWillMount();
            wrapper.instance().getItem();
            wrapper.instance().getLists();
            wrapper.instance().onSubmit();
            
        });
    it('render the two inputs', () =>{
            expect(wrapper.find(<Form.Input id="newamount"/>));
            expect(wrapper.find(<Form.Input id="newquantity"/>));
            expect(shallowToJson(wrapper)).toMatchSnapshot();
        });
})