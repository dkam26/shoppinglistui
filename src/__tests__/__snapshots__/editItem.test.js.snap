// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit item component render the two inputs 1`] = `
<div>
  <Container>
    <Segment>
      <Menu
        secondary={true}
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <MenuItem
          name="SHOPPINGLIST"
          style={
            Object {
              "color": "white",
            }
          }
        />
        <MenuMenu
          position="right"
        >
          <MenuItem
            name="logout"
            style={
              Object {
                "color": "white",
              }
            }
          />
        </MenuMenu>
      </Menu>
    </Segment>
    <Segment>
      <Menu
        secondary={true}
      >
        <MenuMenu>
          <MenuMenu
            position="right"
          >
            <MenuItem>
              <Button
                as="button"
                content="Back"
                icon="left chevron"
                labelPosition="left"
                onClick={[Function]}
              />
            </MenuItem>
          </MenuMenu>
        </MenuMenu>
        <MenuMenu>
          <MenuItem>
            <h3
              style={
                Object {
                  "marginLeft": "308px",
                }
              }
            >
              Edit  
               item
            </h3>
          </MenuItem>
        </MenuMenu>
      </Menu>
    </Segment>
    <div
      className="login-form"
    >
      <style>
        
          body &gt; div,
          body &gt; div &gt; div,
          body &gt; div &gt; div &gt; div.login-form {
            height: 100%;
          }
        
      </style>
      <Grid
        style={
          Object {
            "height": "100%",
          }
        }
        textAlign="center"
        verticalAlign="middle"
      >
        <GridColumn
          style={
            Object {
              "maxWidth": 450,
            }
          }
        >
          <Form
            as="form"
            size="large"
          >
            <Segment
              stacked={true}
            >
              Quantity: 
              <FormInput
                as={[Function]}
                control={[Function]}
                fluid={true}
                id="newquantity"
                name="newquantity"
                onChange={[Function]}
                required={true}
                type="number"
                value=""
              />
              Amount:
              <FormInput
                as={[Function]}
                control={[Function]}
                fluid={true}
                id="newamount"
                name="newamount"
                onChange={[Function]}
                required={true}
                type="number"
                value=""
              />
              <Button
                as="button"
                color="blue"
                fluid={true}
                id="editItem"
                onClick={[Function]}
                size="large"
              >
                Edit Item
              </Button>
            </Segment>
          </Form>
        </GridColumn>
      </Grid>
    </div>
  </Container>
  <_class
    options={Object {}}
  />
</div>
`;

exports[`edit item component should edit item 1`] = `
<div>
  <Container>
    <Segment>
      <Menu
        secondary={true}
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <MenuItem
          name="SHOPPINGLIST"
          style={
            Object {
              "color": "white",
            }
          }
        />
        <MenuMenu
          position="right"
        >
          <MenuItem
            name="logout"
            style={
              Object {
                "color": "white",
              }
            }
          />
        </MenuMenu>
      </Menu>
    </Segment>
    <Segment>
      <Menu
        secondary={true}
      >
        <MenuMenu>
          <MenuMenu
            position="right"
          >
            <MenuItem>
              <Button
                as="button"
                content="Back"
                icon="left chevron"
                labelPosition="left"
                onClick={[Function]}
              />
            </MenuItem>
          </MenuMenu>
        </MenuMenu>
        <MenuMenu>
          <MenuItem>
            <h3
              style={
                Object {
                  "marginLeft": "308px",
                }
              }
            >
              Edit  
               item
            </h3>
          </MenuItem>
        </MenuMenu>
      </Menu>
    </Segment>
    <div
      className="login-form"
    >
      <style>
        
          body &gt; div,
          body &gt; div &gt; div,
          body &gt; div &gt; div &gt; div.login-form {
            height: 100%;
          }
        
      </style>
      <Grid
        style={
          Object {
            "height": "100%",
          }
        }
        textAlign="center"
        verticalAlign="middle"
      >
        <GridColumn
          style={
            Object {
              "maxWidth": 450,
            }
          }
        >
          <Form
            as="form"
            size="large"
          >
            <Segment
              stacked={true}
            >
              Quantity: 
              <FormInput
                as={[Function]}
                control={[Function]}
                fluid={true}
                id="newquantity"
                name="newquantity"
                onChange={[Function]}
                required={true}
                type="number"
                value=""
              />
              Amount:
              <FormInput
                as={[Function]}
                control={[Function]}
                fluid={true}
                id="newamount"
                name="newamount"
                onChange={[Function]}
                required={true}
                type="number"
                value=""
              />
              <Button
                as="button"
                color="blue"
                fluid={true}
                id="editItem"
                onClick={[Function]}
                size="large"
              >
                Edit Item
              </Button>
            </Segment>
          </Form>
        </GridColumn>
      </Grid>
    </div>
  </Container>
  <_class
    options={Object {}}
  />
</div>
`;

exports[`edit item component should render correctly 1`] = `
<div>
  <Container>
    <Segment>
      <Menu
        secondary={true}
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <MenuItem
          name="SHOPPINGLIST"
          style={
            Object {
              "color": "white",
            }
          }
        />
        <MenuMenu
          position="right"
        >
          <MenuItem
            name="logout"
            style={
              Object {
                "color": "white",
              }
            }
          />
        </MenuMenu>
      </Menu>
    </Segment>
    <Segment>
      <Menu
        secondary={true}
      >
        <MenuMenu>
          <MenuMenu
            position="right"
          >
            <MenuItem>
              <Button
                as="button"
                content="Back"
                icon="left chevron"
                labelPosition="left"
                onClick={[Function]}
              />
            </MenuItem>
          </MenuMenu>
        </MenuMenu>
        <MenuMenu>
          <MenuItem>
            <h3
              style={
                Object {
                  "marginLeft": "308px",
                }
              }
            >
              Edit  
               item
            </h3>
          </MenuItem>
        </MenuMenu>
      </Menu>
    </Segment>
    <div
      className="login-form"
    >
      <style>
        
          body &gt; div,
          body &gt; div &gt; div,
          body &gt; div &gt; div &gt; div.login-form {
            height: 100%;
          }
        
      </style>
      <Grid
        style={
          Object {
            "height": "100%",
          }
        }
        textAlign="center"
        verticalAlign="middle"
      >
        <GridColumn
          style={
            Object {
              "maxWidth": 450,
            }
          }
        >
          <Form
            as="form"
            size="large"
          >
            <Segment
              stacked={true}
            >
              Quantity: 
              <FormInput
                as={[Function]}
                control={[Function]}
                fluid={true}
                id="newquantity"
                name="newquantity"
                onChange={[Function]}
                required={true}
                type="number"
                value=""
              />
              Amount:
              <FormInput
                as={[Function]}
                control={[Function]}
                fluid={true}
                id="newamount"
                name="newamount"
                onChange={[Function]}
                required={true}
                type="number"
                value=""
              />
              <Button
                as="button"
                color="blue"
                fluid={true}
                id="editItem"
                onClick={[Function]}
                size="large"
              >
                Edit Item
              </Button>
            </Segment>
          </Form>
        </GridColumn>
      </Grid>
    </div>
  </Container>
  <_class
    options={Object {}}
  />
</div>
`;
